src/kitovu/sync/syncing.py

```diff
@@ -30,16 +30,17 @@ def _find_plugin(pluginname: str) -> syncplugin.AbstractSyncPlugin:
     return plugin
 
 
-def start_all(config_file: typing.Optional[pathlib.Path]) -> None:
+def start_all(config_file: typing.Optional[pathlib.Path], reporter: utils.AbstractReporter) -> None:
     """Sync all files with the given configuration file."""
     settings = Settings.from_yaml_file(config_file)
     for _plugin_key, connection_settings in sorted(settings.connections.items()):
-        start(connection_settings)
+        start(connection_settings, reporter)
 
 
-def start(connection_settings: ConnectionSettings) -> None:
+def start(connection_settings: ConnectionSettings, reporter: utils.AbstractReporter) -> None:
     """Sync files with the given plugin and username."""
     plugin = _find_plugin(connection_settings.plugin_name)
+    plugin.reporter = reporter
```

Comment:
It's bad practice (probably more than in Ruby :laughing:) to monkeypatch things like this. Instead, we should probably pass the reporter to `_find_plugin`, and that can pass it as `invoke_args` to `stevedore.driver.DriverManager` so `stevedore` passes it to the plugin's constructor ([docs](https://docs.openstack.org/stevedore/latest/reference/index.html#stevedore.driver.DriverManager))